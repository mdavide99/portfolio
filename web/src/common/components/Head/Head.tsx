import Link from "next/link";
import { NavBar } from "../../elements/Navigation";

function Head() {
  return (
    <div id="head" className="container flex items-center justify-between mx-auto px-4 h-20 mb-10">
      <div id="logo" className="flex items-center text-3xl sm:text-4xl font-extrabold group">
        <Link href={"/"} passHref={true}>
          <a>
            <span className="group-hover:text-neutral-50 dark:group-hover:text-neutral-900 group-hover:transition-colors duration-1000 delay-75 group-hover:duration-500 group-hover:ease-linear">
              {"// "}
            </span>
            <span className="group-hover:text-blue-500 dark:group-hover:text-fuchsia-500 group-hover:transition-colors duration-1000 delay-75 group-hover:duration-500 group-hover:ease-linear tracking-wide">
              Davide
            </span>
          </a>
        </Link>
      </div>

      <NavBar />
    </div>
  );
}

export { Head };
