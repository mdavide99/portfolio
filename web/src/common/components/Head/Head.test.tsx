import { Head } from "./Head";
import { render } from "@testing-library/react";

jest.mock("next/router", () => ({
    useRouter() {
        return {
            route: "/",
            pathname: "",
            query: "",
            asPath: "",
			events:{on: jest.fn}
        };
    },
}));
describe('Head',()=>{

  it('renders Head correctly', () => {

		const { container } = render(
		<Head/>
		);
		expect(container).toMatchSnapshot();
	});
})