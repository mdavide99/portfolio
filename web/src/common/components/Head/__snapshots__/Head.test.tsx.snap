// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Head renders Head correctly 1`] = `
<div>
  <div
    class="container flex items-center justify-between mx-auto px-4 h-20 mb-10"
    id="head"
  >
    <div
      class="flex items-center text-3xl sm:text-4xl font-extrabold group"
      id="logo"
    >
      <a
        href="/"
      >
        <span
          class="group-hover:text-neutral-50 dark:group-hover:text-neutral-900 group-hover:transition-colors duration-1000 delay-75 group-hover:duration-500 group-hover:ease-linear"
        >
          // 
        </span>
        <span
          class="group-hover:text-blue-500 dark:group-hover:text-fuchsia-500 group-hover:transition-colors duration-1000 delay-75 group-hover:duration-500 group-hover:ease-linear tracking-wide"
        >
          Davide
        </span>
      </a>
    </div>
    <button>
      <svg
        class="sm:hidden dark:text-neutral-50 text-xl hover:text-blue-500 hover:dark:text-fuchsia-500 hover:transition-color duration-500"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
        />
      </svg>
    </button>
    <div
      class="fixed inset-x-0 bottom-0 top-0 pb-40 sm:pb-0 sm:static flex flex-col justify-center sm:flex-row items-center bg-neutral-50 dark:bg-neutral-900 sm:bg-transparent hidden sm:block "
    >
      <button>
        <svg
          class="absolute top-7 right-4 sm:hidden dark:text-neutral-50 text-xl hover:text-blue-500 hover:dark:text-fuchsia-500 hover:transition-color duration-500"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 352 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          />
        </svg>
      </button>
      <a
        class="text-4xl sm:text-base my-3 sm:my-0 mx-5 text-gray-500 dark:text-neutral-100 hover:text-gray-900 hover:dark:text-white hover:transition-color hover:underline decoration-blue-500 dark:decoration-fuchsia-500 decoration-4 sm:decoration-2 underline-offset-8"
        href="/"
      >
        about
      </a>
      <a
        class="text-4xl sm:text-base my-3 sm:my-0 mx-5 text-gray-500 dark:text-neutral-100 hover:text-gray-900 hover:dark:text-white hover:transition-color hover:underline decoration-blue-500 dark:decoration-fuchsia-500 decoration-4 sm:decoration-2 underline-offset-8"
        href="/projects"
      >
        projects
      </a>
      <a
        class="text-4xl sm:text-base my-3 sm:my-0 mx-5 text-gray-500 dark:text-neutral-100 hover:text-gray-900 hover:dark:text-white hover:transition-color hover:underline decoration-blue-500 dark:decoration-fuchsia-500 decoration-4 sm:decoration-2 underline-offset-8"
        href="/contact"
      >
        contact
      </a>
    </div>
  </div>
</div>
`;
